src/main/java/org/openstreetmap/atlas/checks/validation/intersections/IntersectionAtDifferentLevelsCheck.java

```diff
@@ -0,0 +1,123 @@
+package org.openstreetmap.atlas.checks.validation.intersections;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.checks.utility.CommonMethods;
+import org.openstreetmap.atlas.geography.Location;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.items.Node;
+import org.openstreetmap.atlas.tags.AreaTag;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.LayerTag;
+import org.openstreetmap.atlas.tags.RailwayTag;
+import org.openstreetmap.atlas.tags.filters.TaggableFilter;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * This check is to detect and flag a node when it is a non-terminal intersection node between two
+ * ways which have different layer tag values.
+ *
+ * @author Vladimir Lemberg
+ */
+public class IntersectionAtDifferentLevelsCheck extends BaseCheck<Long>
+{
+    // Instructions
+    private static final String INSTRUCTION_FORMAT = "The Node id {0,number,#} connects two Ways at different levels.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Collections
+            .singletonList(INSTRUCTION_FORMAT);
+    private static final String GREAT_SEPARATION_FILTER_DEFAULT = "bridge->yes|tunnel->yes|embankment->yes|cutting->yes|ford->yes";
```

Comment:
I think it best to make the default for this an empty string. Then add this string as a recommended filter to reduce the flags to high priority ways if wanted.